File: Suraj/Sorting/Bubble_sort.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> bubbleSort(vector<int>& nums) {
    int n = nums.size();
    for(int i=0;i<n;i++){
        for(int j=0;j<n-1;j++){
            if(nums[j]>nums[j+1]){
                swap(nums[j],nums[j+1]);
            }
        }
    }
    return nums;
    }
};

int main() {
    // Create an instance of solution 
    Solution solution;
    
    vector<int> nums = {7, 4, 1, 5, 3};
    
    cout << "Before Using Bubble Sort: " << endl;
    for (int num : nums) {
        cout << num << " ";
    }
    cout << endl;

    // Function call for Bubble Sort
    nums = solution.bubbleSort(nums);

    cout << "Array After Using Bubble Sort: " << endl;
    for (int num : nums) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: python/sorting/bubbleSort.py
class Solution:
    def bubbleSort(self, nums):
        n=len(nums)
        for i in range(n-1):
            for j in range(n-i-1):
                if(nums[j]>nums[j+1]):
                    nums[j],nums[j+1]=nums[j+1],nums[j]
        return nums

if __name__=="__main__":
    nums=[64, 25, 12, 22, 11]
    s=Solution()
    print(s.bubbleSort(nums))

    # by ANSH PATERIYA